internal/surfer/gcloud/builder.go

```diff
@@ -0,0 +1,403 @@
+		param.HelpText = rule.HelpText.Brief
+	} else {
+		// TODO(https://github.com/googleapis/librarian/issues/3033): improve default help text inference
+		param.HelpText = fmt.Sprintf("Value for the `%s` field.", strcase.ToKebab(field.Name))
+	}
+	return param, nil
+}
+
+// newPrimaryResourceParam creates the main positional resource argument for a command.
+// This is the argument that represents the resource being acted upon (e.g., the instance name).
+func newPrimaryResourceParam(field *api.Field, method *api.Method, model *api.API, _ *Config, service *api.Service) Param {
+	// We first need to get the full resource definition for the method.
+	resource := utils.GetResourceForMethod(method, model)
+	var segments []api.PathSegment
+	if resource != nil && len(resource.Patterns) > 0 {
+		segments = resource.Patterns[0]
+	}
+
+	// We construct the gcloud collection path from the resource's pattern string.
+	// Example: `projects/{project}/locations/{location}/instances/{instance}` -> `projects.locations.instances`
+	collectionPath := utils.GetCollectionPathFromSegments(segments)
+	hostParts := strings.Split(service.DefaultHost, ".")
+	shortServiceName := hostParts[0]
+
+	// We determine the singular name of the resource.
+	// For `Create` methods, this comes from the `_id` field. For others, it's the `name` field.
+	resourceName := strcase.ToSnake(strings.TrimSuffix(field.Name, "_id"))
+	if field.Name == "name" {
+		resourceName = utils.GetSingularFromSegments(segments)
+	}
+
+	// We generate a helpful help text based on whether the command is a `Create` command or not.
+	helpText := fmt.Sprintf("The %s to create.", resourceName)
+	if !strings.HasPrefix(method.Name, "Create") {
+		helpText = fmt.Sprintf("The %s to operate on.", resourceName)
+	}
+
+	// We assemble the final `Param` struct with all the necessary information for a primary resource.
+	return Param{
+		HelpText:          helpText,
+		IsPositional:      true,
+		IsPrimaryResource: true,
+		Required:          true,
+		RequestIDField:    strcase.ToLowerCamel(field.Name),
+		ResourceSpec: &ResourceSpec{
+			Name:                  resourceName,
+			PluralName:            utils.GetPluralResourceNameForMethod(method, model),
+			Collection:            fmt.Sprintf("%s.%s", shortServiceName, collectionPath),
+			DisableAutoCompleters: false,
+			Attributes:            newAttributesFromSegments(segments),
+		},
+	}
+}
+
+// newResourceReferenceSpec creates a ResourceSpec for a field that references
+// another resource type (e.g., a `--network` flag).
+func newResourceReferenceSpec(field *api.Field, model *api.API, _ *Config, service *api.Service) (*ResourceSpec, error) {
+	// We iterate through all the resource definitions in the API model to find the
+	// one that matches the type of our resource reference.
+	for _, def := range model.ResourceDefinitions {
+		if def.Type == field.ResourceReference.Type {
+			if len(def.Patterns) == 0 {
```

Comment:
Nit: I hate pedantic coding rules generally but this rule of thumb I think is helpful. If you have more than two layers of nesting, it's usually best to use guard statement or helper functions. TOTT has a rule for it